actions/views/custom_status.ts

```diff
@@ -0,0 +1,45 @@
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {updateMe} from 'mattermost-redux/actions/users';
+import {DispatchFunc, GetStateFunc} from 'mattermost-redux/types/actions';
+
+import {CustomStatus} from 'types/store/custom_status';
+
+export function updateUserCustomStatus(newCustomStatus: CustomStatus) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const userProps = Object.assign({}, getCurrentUser(getState()).props);
```

Comment:
Why do we need to use `Object.assign` here?

Reply:
I have updated this as discussed